### PR TITLE
PLAT-115 remove mds db policy artifacts

### PR DESCRIPTION
**Pull request by nessur**
## 📚 Purpose
*[removed mds-db/policies, moved types to mds-policy-service]*

## 👌 Resolves:
- [ ] ✨ PLAT-115

## 📦 Impacts:
*[mds-db]*
*[mds-test-data]*
*[mds-types]*
*[mds-utils]*
*[mds-compliance-api]*
*[mds-compliance-engine]*
*[mds-compliance-service]*
*[mds-policy]*
*[mds-policy-author]*
*[mds-policy-service]*

## ✅ PR Checklist
- [ ] build a release

### CODE REVIEW

File: packages/mds-policy-service/test_data/policies.ts [hunk 1]
```diff
@@ -1,4 +1,18 @@
-import { Policy } from '@mds-core/mds-types'
+/**
+ * Copyright 2021 russenguggemos
```
Comment (drtyh2o) [inline packages/mds-policy-service/test_data/policies.ts:2]: Add the following property to `vscode/settings.json` in this repo and change here.
```json
"licenser.author": "City of Los Angeles"
```

**Review by drtyh2o — APPROVED** [review]
Outstanding!

File: packages/mds-policy-service/test_data/policies.ts (hunk 1 above)
Comment (nessur) [reply packages/mds-policy-service/test_data/policies.ts:2]: thanks!

File: packages/mds-compliance-engine/engine/count_processors.ts [hunk 2]
```diff
@@ -60,13 +54,13 @@ export function isCountRuleMatch(
...
-  policy: Policy,
+  policy: CountPolicy,
   events: (VehicleEvent & { telemetry: Telemetry })[],
   geographies: Geography[],
   devices: { [d: string]: Device }
 ): ComplianceEngineResult | undefined {
   if (getPolicyType(policy) !== RULE_TYPES.count) {
-    throw new UnsupportedTypeError(`${getPolicyType(policy)} submitted to count processor`)
+    throw new UnsupportedTypeError(`${getPolicyType(policy) as string} submitted to count processor`)
   }
```
Comment (avatarneil) [inline packages/mds-compliance-engine/engine/count_processors.ts:64]: This block should be logically unnecessary now, assuming that what's fed in here is definitely a `CountPolicy` (which is what the types now indicate)

File: packages/mds-compliance-engine/engine/helpers.ts [hunk 3]
```diff
@@ -1,38 +1,60 @@
...
+}
+
+export function isSpeedPolicy(policy: PolicyDomainModel): policy is SpeedPolicy {
+  return policy.rules[0].rule_type === 'speed'
+}
+
+export function isTimePolicy(policy: PolicyDomainModel): policy is TimePolicy {
+  return policy.rules[0].rule_type === 'time'
+}
+
```
Comment (avatarneil) [inline packages/mds-compliance-engine/engine/helpers.ts:50]: Maybe out of scope for this PR, but I'd love this to go and run this assertion across all rules. Something along the lines of `export const isCountPolicy = policy.rules.reduce<boolean>((acc, { rule_type }) => acc ? rule_type === 'count' : acc), true)`

**Review by avatarneil — APPROVED** [review]

File: packages/mds-compliance-engine/engine/helpers.ts (hunk 3 above)
Comment (drtyh2o) [reply packages/mds-compliance-engine/engine/helpers.ts:50]: Or, just `export const isCountPolicy = policy.rules.every(({ rule_type }) => rule_type === ‘count’)` … or even a function that accepts a `rule_type` as a parameter.

Comment (avatarneil) [reply packages/mds-compliance-engine/engine/helpers.ts:50]: I forgot that `.every` was a thing! That'd be much more readable than my impl for sure :)

**Pull request merged**